src/TBuffer.cpp

```diff
@@ -2441,75 +2441,113 @@ inline int TBuffer::wrap(int startLine)
     if (static_cast<int>(buffer.size()) < startLine || startLine < 0) {
         return 0;
     }
+
     std::queue<std::deque<TChar>> queue;
     QStringList tempList;
     QStringList timeList;
     QList<bool> promptList;
     int lineCount = 0;
+    const QString lineBreaks = qsl(",.- ，。　()");
+
+    const auto hostFont = mpHost->getDisplayFont();
+    int mFontWidth = QFontMetrics(hostFont).averageCharWidth();
+    int wrapByPixel = mWrapAt * mFontWidth;
+    
+    // Loop from current line up to the last line sitting in buffer
     for (int i = startLine, total = static_cast<int>(buffer.size()); i < total; ++i) {
+        // Capture isPrompt
         bool isPrompt = promptBuffer[i];
         std::deque<TChar> newLine;
         QString lineText = "";
         QString time = timeBuffer[i];
-        int indent = 0;
-        if (static_cast<int>(buffer[i].size()) >= mWrapAt) {
-            for (int i3 = 0; i3 < mWrapIndent; ++i3) {
-                TChar pSpace;
-                newLine.push_back(pSpace);
-                lineText.append(" ");
-            }
-            indent = mWrapIndent;
-        }
-        int lastSpace = 0;
+        // Index where wrap should happen
         int wrapPos = 0;
+        // Number of character in this line
         int length = buffer[i].size();
+
+        // If number of character is zero, this is just blank line, which actually means the for loop after the if statement won't happen
         if (length == 0) {
             tempList.append(QString());
             std::deque<TChar> emptyLine;
             queue.push(emptyLine);
             timeList.append(time);
+            promptList.append(false);
         }
-        for (int i2 = 0, total = static_cast<int>(buffer[i].size()); i2 < total;) {
-            if (length - i2 > mWrapAt - indent) {
-                wrapPos = calculateWrapPosition(i, i2, i2 + mWrapAt - indent);
-                lastSpace = qMax(0, wrapPos);
-            } else {
-                lastSpace = 0;
-            }
-            int wrapPosition = (lastSpace) ? lastSpace : (mWrapAt - indent);
-            for (int i3 = 0; i3 < wrapPosition; ++i3) {
-                if (lastSpace > 0) {
-                    if (i2 > lastSpace) {
-                        break;
-                    }
-                }
-                if (i2 >= static_cast<int>(buffer[i].size())) {
-                    break;
-                }
-                if (lineBuffer[i].at(i2) == '\n') {
-                    i2++;
-                    break;
-                }
+
+        // Track whether there is data that needs to be appended
+        bool hasContent = false;
+
+        // Track the current width in pixel
+        int lineWidth = QFontMetrics(hostFont).horizontalAdvance(lineBuffer[i]);
```

Comment:
```suggestion
        int lineWidth = QFontMetrics(hostFont).horizontalAdvance(lineBuffer.at(i));
```

Reply:
I've made the changes, thanks, solid suggestion :)